webapp/channels/src/entry.tsx

```diff
@@ -77,3 +77,26 @@ appendOnDOMContentLoadedEvent(() => {
     // Do the pre-render setup and call renderReactRootComponent when done
     preRenderSetup(renderReactRootComponent);
 });
+
+const registerServiceWorker = async () => {
+    if ('serviceWorker' in navigator) {
+        try {
+            const registration = await navigator.serviceWorker.register('/sw.js');
+            if (registration.installing) {
```

Comment:
This API is confusing me a bit. It seems like `registration` contains 3 different potential references to the same `ServiceWorker` object instead of having a single `registration.serviceWorker` and some `registration.registrationState` field. Is that right?

I'm trying to understand what we might do with the object returned by `navigator.serviceWorker.register` at this point other than just logging it

Reply:
Just logging it. It's because the first time you load it, it's in installing phase. Only the next time you reload, it will be active.